src/cobra/core/reaction.py

```diff
@@ -908,16 +916,13 @@ def copy(self) -> "Reaction":
         self._model = None
         for i in self._metabolites:
             i._model = None
-        for i in self._genes:
-            i._model = None
         # now we can copy
         new_reaction = deepcopy(self)
+        new_reaction.gpr = self.gpr.copy()
         # restore the references
         self._model = model
         for i in self._metabolites:
             i._model = model
-        for i in self._genes:
```

Comment:
This has to stay for SBML compatibility. We do explicitly allow genes to be part of a reaction even if the reaction has no GPR (and even if they don't appear in the GPR).

Reply:
Hmm.
I think one discussion with @Midnighter gave me the impression that GPR should be the only thing updated, so gene_reaction_rule and genes are always derived from it. That led to the function update_genes_from_gpr()
If you set up a model with a reaction that has genes and no GPR, when you run update_genes_from_gpr() you'll wipe out the genes field of that reaction. update_genes_from_gpr is called in various time points, including renaming genes and others...

So maybe we should think on how to do update without wiping some genes? Do you have any test case for this SBML compatibility?

Reply:
Though you only delete the genes from the reaction, not the model, right? This may be okay but it leads to some odd behavior. I tracked it in #1185 to get some more input.

Reply:
Well. Not exactly.
If the gene has no reactions after removal from this reaction, the function tries to remove it from the model. I could change that behavior, but see my comments on #1185